components/Themes/src/MDCColorScheme.m

```diff
@@ -16,6 +16,76 @@
 
 #import "MDCColorScheme.h"
 
-@implementation MDCColorScheme
+@implementation MDCBasicColorScheme
+
+- (nonnull instancetype)initWithPrimaryColor:(nonnull UIColor *)primaryColor
+                           primaryLightColor:(nonnull UIColor *)primaryLightColor
+                            primaryDarkColor:(nonnull UIColor *)primaryDarkColor
+                              secondaryColor:(nonnull UIColor *)secondaryColor
+                         secondaryLightColor:(nonnull UIColor *)secondaryLightColor
+                          secondaryDarkColor:(nonnull UIColor *)secondaryDarkColor {
+  self = [super init];
+  if (self) {
+    _primaryColor = primaryColor;
+    _primaryLightColor = primaryLightColor;
+    _primaryDarkColor = primaryDarkColor;
+    _secondaryColor = secondaryColor;
+    _secondaryLightColor = secondaryLightColor;
+    _secondaryDarkColor = secondaryDarkColor;
+  }
+  return self;
+}
+
+- (nonnull instancetype)initWithPrimaryColor:(nonnull UIColor *)primaryColor {
+  UIColor *primaryLightColor = [self lighterColorForColor:primaryColor];
+  UIColor *primaryDarkColor = [self darkerColorForColor:primaryColor];
+  return [self initWithPrimaryColor:primaryColor
+                  primaryLightColor:primaryLightColor
+                   primaryDarkColor:primaryDarkColor
+                     secondaryColor:primaryColor
+                secondaryLightColor:primaryLightColor
+                 secondaryDarkColor:primaryDarkColor];
+}
+
+- (nonnull instancetype)initWithPrimaryColor:(nonnull UIColor *)primaryColor
+                           primaryLightColor:(nonnull UIColor *)primaryLightColor
+                            primaryDarkColor:(nonnull UIColor *)primaryDarkColor {
+  return [self initWithPrimaryColor:primaryColor
+                  primaryLightColor:primaryLightColor
+                   primaryDarkColor:primaryDarkColor
+                     secondaryColor:primaryColor
+                secondaryLightColor:primaryLightColor
+                 secondaryDarkColor:primaryDarkColor];
+}
+
+- (nonnull instancetype)initWithPrimaryColor:(nonnull UIColor *)primaryColor
+                              secondaryColor:(nonnull UIColor *)secondaryColor {
+  return [self initWithPrimaryColor:primaryColor
+                  primaryLightColor:[self lighterColorForColor:primaryColor]
+                   primaryDarkColor:[self darkerColorForColor:primaryColor]
+                     secondaryColor:secondaryColor
+                secondaryLightColor:[self lighterColorForColor:secondaryColor]
+                 secondaryDarkColor:[self darkerColorForColor:secondaryColor]];
+}
+
+- (UIColor *)lighterColorForColor:(UIColor *)color {
+  CGFloat r, g, b, a;
+  if ([color getRed:&r green:&g blue:&b alpha:&a])
+    return [UIColor colorWithRed:MIN(r + 0.2, 1.0)
```

Comment:
Whoa. Please use `+UIColor (BOOL)getHue:(CGFloat *)hue saturation:(CGFloat *)saturation brightness:(CGFloat *)brightness alpha:(CGFloat *)alpha` and bump the brightness and then convert back. Same below.